src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobContentId.cs

```diff
@@ -66,16 +66,23 @@ public static unsafe BlobContentId FromHash(byte[] hashCode)
                 throw new ArgumentException(SR.Format(SR.HashTooShort, minHashSize), nameof(hashCode));
             }
 
-            Guid guid = default(Guid);
-            byte* guidPtr = (byte*)&guid;
-            for (var i = 0; i < BlobUtilities.SizeOfGuid; i++)
-            {
-                guidPtr[i] = hashCode[i];
-            }
+            // extract guid components from input data
+            uint a = ((uint)hashCode[3] << 24 | (uint)hashCode[2] << 16 | (uint)hashCode[1] << 8 | hashCode[0]);
+            ushort b = (ushort)((ushort)hashCode[5] << 8 | (ushort)hashCode[4]);
+            ushort c = (ushort)((ushort)hashCode[7] << 8 | (ushort)hashCode[6]);
+            byte d = hashCode[8];
+            byte e = hashCode[9];
+            byte f = hashCode[10];
+            byte g = hashCode[11];
+            byte h = hashCode[12];
+            byte i = hashCode[13];
+            byte j = hashCode[14];
+            byte k = hashCode[15];
 
             // modify the guid data so it decodes to the form of a "random" guid ala rfc4122
-            guidPtr[7] = (byte)((guidPtr[7] & 0x0f) | (4 << 4));
-            guidPtr[8] = (byte)((guidPtr[8] & 0x3f) | (2 << 6));
+            c = (ushort)((c & 0x0fff) | (4 << 12));
+            d = (byte)((d & 0x3f) | (2 << 6));
+            Guid guid = new Guid((int)a, (short)b, (short)c, d, e, f, g, h, i, j, k);
```

Comment:
```suggestion
            Guid guid = new Guid((int)a, (short)b, (short)c,
                hashCode[8], hashCode[9], hashCode[10], hashCode[11],
                hashCode[12], hashCode[13], hashCode[14], hashCode[15]);
```
Would this look better?

Reply:
This would be incorrect, since `d` needs to be modified to comply with the RFC4122 rule.  I guess the remaining 7 bytes could be done without temp variables, but I'm not really sure this looks better ...

Reply:
Oh, ok. Ignore this suggestion then.